src/openrct2-ui/windows/ServerList.cpp

```diff
@@ -267,22 +268,17 @@ class ServerListWindow final : public Window
         if (text.empty())
             return;
 
-        std::string temp = static_cast<std::string>(text);
+        u8string temp = static_cast<u8string>(text);
 
         switch (widgetIndex)
         {
             case WIDX_PLAYER_NAME_INPUT:
                 if (_playerName == text)
                     return;
 
-                text.copy(_playerName, sizeof(_playerName));
-
-                // Don't allow empty player names
-                if (_playerName[0] != '\0')
-                {
-                    gConfigNetwork.PlayerName = _playerName;
-                    ConfigSaveDefault();
-                }
+                _playerName = temp;
```

Comment:
This will leak the old pointer. You must reset the widget pointer after this

Reply:
Yea, I had thought I solved this by setting the underlying c_str to be the same size as the max name length so on it would copy and not proc a realloc, but I thought about it, and this would be vendor implementation specific and is not defined in the standard. I think the approach you suggest is the safest: [3d02334](https://github.com/OpenRCT2/OpenRCT2/pull/20004/commits/3d02334f2dc3124cf3cc737ee50f49aaa3a849de).